meta-buildpacks/nodejs/buildpack.toml

```diff
@@ -60,5 +60,11 @@ optional = true
 id = "heroku/nodejs-npm-install"
 version = "2.2.0"
 
+[[order]]
+
+[[order.group]]
+id = "heroku/nodejs-engine"
```

Comment:
I'm wondering if this catch-all group should be it's own buildpack?  If the logic is in the node-engine, this will force the need for a `package.json` in the functions composite buildpack too, right?

Reply:
Yes, you are right. It would have impacted the functions composite buildpack as it was. I've moved the missing lockfile logic into the `heroku/nodejs-npm-install` buildpack instead.

Reply:
It would be great if we could get rid of the Bash based npm buildpack sooner rather than later (we can then clean up some of the non-libcnb.rs buildpack handling in the release automation etc, amongst other things).

To unblock being able to remove the Bash buildpack without enforcing lockfiles on functions customers, what do you think about making the check conditional on the project not being a functions project? Whilst adding such a conditional isn't ideal, it (a) seems better than having to keep the old Bash buildpack around, (b) it could probably be removed sooner than Functions EOL so long as we handle the Functions changelog/comms side and have let the deprecation notice show up in functions build logs for a bit.

If you take that approach, then the lockfile check could still be run in the npm engine buildpack (if desired; I have no preference either way).

Reply:
Luckily, @colincasey has already added comms about the deprecation in the bash buildpack. So, we can move functions to `heroku/nodejs-npm-install` as soon as we think it's been long enough warning perioud. If we try to move functions to `heroku/nodejs-npm-install` any sooner, I think that buildpack would need support for lockfile-less installs, which I'm hoping to avoid.

We could retire the source for the bash buildpack earlier if we wanted, by referencing the bash buildpack artifact on dockerhub in the functions meta-buildpack (that would presume it won't need changes). Or we could move it into it's own repo.

Reply:
We could also do what we originally did not want to do and add some metadata to the buildpack plan from the `nodejs-function-invoker` CNB that `npm-install` can use to determine if it should:
- allow installing modules with no lockfile
- run the build hooks in the same way as our classic Node buildpack

It's not ideal but it would allow us to at least remove the bash CNB entirely at the cost of having to support a bit of deprecated logic.